### DISCUSSION TITLE
How to upload files like images

### DISCUSSION THREAD

**Discussion by braunpa** [heftapp/graphql_codegen#301] [Q&A] [answered]
I would like to upload files but the graphql_codegen is transforming Upload! to String. Is there a solution to uploading a file?

**Comment by budde377** [comment, marked as answer]
You need to map the `Upload` scalar to something that `graphql_flutter` can handle, e.g, you need to add a [custom scalar](https://github.com/heftapp/graphql_codegen/tree/main/packages/graphql_codegen#custom-scalars) and then map it to a `MultipartFile` (see [here](https://github.com/zino-hofmann/graphql-flutter/blob/main/packages/graphql/README.md#graphql-upload))
